packages/origin-247-energy-api/src/proof-request/repositories/proof-request.repository.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { In, Repository } from 'typeorm';
+import { Reading } from '../../util';
+import { ProofRequest, ProofRequestState, proofRequestTableName } from '../proof-request.entity';
+
+interface CreateNewRequestPayload {
+    deviceId: string;
+    reading: Reading;
+}
+
+interface SaveProcessingErrorParams {
+    error: string;
+    requestIds: number[];
+}
+
+@Injectable()
+export class ProofRequestsRepository {
+    constructor(
+        @InjectRepository(ProofRequest)
+        private repository: Repository<ProofRequest>
+    ) {}
+
+    public async createNewRequest(...payload: CreateNewRequestPayload[]): Promise<void> {
+        await this.repository.save(
+            payload.map((p) => ({
+                deviceId: p.deviceId,
+                reading: p.reading,
+                state: ProofRequestState.Pending,
+                processError: null
+            }))
+        );
+    }
+
+    public async saveProcessingError(params: SaveProcessingErrorParams): Promise<void> {
+        await this.repository.update(
+            { id: In(params.requestIds) },
+            { processError: params.error, state: ProofRequestState.ProcessingError }
+        );
+    }
+
+    public async startProcessing(requestIds: number[]): Promise<void> {
+        await this.repository.update(
+            { id: In(requestIds) },
+            { state: ProofRequestState.Processing }
+        );
+    }
+
+    public async finishProcessing(requestIds: number[]): Promise<void> {
+        await this.repository.delete({ id: In(requestIds) });
+    }
```

Comment:
And here I'd change for deleteProcessed instead. But it would also be misleading since it doesn't check for state = processed to delete them.